### PR TITLE
Fake taker

### PR DESCRIPTION
**Pull request by uv-orbs**

### CODE REVIEW

File: models/order.go [hunk 1]
```diff
@@ -235,9 +235,18 @@ func (o *Order) Fill(ctx context.Context, fillSize decimal.Decimal) (isFilled bo
 	return o.IsFilled(), nil
 }
 
-func (o *Order) Kill(ctx context.Context, size decimal.Decimal) error {
+func (o *Order) Lock(ctx context.Context, size decimal.Decimal) error {
```
Comment (Luke-Rogerson) [inline models/order.go:238]: Probably a good idea to write a few tests around these new methods. I remember I messed up my original logic and only caught the bug when I wrote a test

File: service/taker.go [hunk 2]
```diff
@@ -63,7 +63,11 @@ func (s *Service) BeginSwap(ctx context.Context, data models.QuoteRes) (models.B
 	// set order fragments as Pending
 	for i := 0; i < len(res.Orders); i++ {
 		// lock frag.Amount as pending per order - no STATUS_PENDING is needed
-		res.Orders[i].SizePending = res.Orders[i].SizePending.Add(res.Fragments[i].OutSize)
+		err := res.Orders[i].Lock(ctx, res.Fragments[i].OutSize)
```
Comment (Luke-Rogerson) [inline service/taker.go:66]: Good idea to add a specific method for this

File: service/taker.go [hunk 3]
```diff
@@ -63,7 +63,11 @@ func (s *Service) BeginSwap(ctx context.Context, data models.QuoteRes) (models.B
...
+		if err != nil {
+			logctx.Error(ctx, "Lock order Failed", logger.Error(err))
```
Comment (Luke-Rogerson) [inline service/taker.go:68]: Log orderId so we can debug

File: service/taker.go [hunk 4]
```diff
@@ -114,16 +118,69 @@ func (s *Service) AbortSwap(ctx context.Context, swapId uuid.UUID) error {
 			logctx.Error(ctx, "Swap fragments should be valid during a revert request", logger.Error(err))
 		} else {
 			// success
-			order.SizePending = order.SizePending.Sub(frag.OutSize)
+			err = order.Unlock(ctx, frag.OutSize)
+			if err != nil {
+				logctx.Error(ctx, "Unlock Failed", logger.Error(err))
```
Comment (Luke-Rogerson) [inline service/taker.go:123]: Log orderId so we can debug

File: service/taker.go [hunk 5]
```diff
@@ -114,16 +118,69 @@ func (s *Service) AbortSwap(ctx context.Context, swapId uuid.UUID) error {
...
+
+	filledOrders := []models.Order{}
+	openOrders := []models.Order{}
+	// validate all pending orders fragments of auction
+	for _, frag := range frags {
+		// get order by ID
+		order, err := s.orderBookStore.FindOrderById(ctx, frag.OrderId, false)
+		// no return during erros as what can be revert, should
+		if err != nil {
+			logctx.Error(ctx, "order not found while reverting a swap", logger.Error(err))
```
Comment (Luke-Rogerson) [inline service/taker.go:155]: Log swapId, maybe orderId so we can debug

File: service/taker.go [hunk 6]
```diff
@@ -114,16 +118,69 @@ func (s *Service) AbortSwap(ctx context.Context, swapId uuid.UUID) error {
...
+			if err != nil {
+				logctx.Error(ctx, "Unlock Failed", logger.Error(err))
+				return err
+			}
 			orders = append(orders, *order)
 		}
 	}
 	// store orders
-	err = s.orderBookStore.StoreOpenOrders(ctx, orders)
+	err = s.orderBookStore.StoreFilledOrders(ctx, orders)
```
[the remark this answers is not in the document]
Comment (uv-orbs) [reply service/taker.go:130]: thats the heart of this, if we have multiple frags. most of them are fully filled, and the last one is partial hence stays alive

File: transport/rest/handler.go [hunk 7]
```diff
@@ -142,4 +142,7 @@ func (h *Handler) initTakerRoutes(getUserByApiKey middleware.GetUserByApiKeyFunc
 	// IN: txHash, SwapID given in /swap
 	// Notifies order book to start tracking the state of the tx (discuss events or based on txHash)
 	// takerApi.HandleFunc("/txsent/{swapId}", h.txSent).Methods("POST")
+
+	// ------- DEBUG -------
+	takerApi.HandleFunc("/fake-fill", h.fakeFill).Methods("POST")
```
[the remark this answers is not in the document]
Comment (uv-orbs) [reply transport/rest/handler.go:147]: sure i forgot about it

**Pull request merged**